src/OrchardCore.Modules/OrchardCore.Contents/AdminMenu.cs

```diff
@@ -41,7 +41,7 @@ public async Task BuildNavigationAsync(string name, NavigationBuilder builder)
                 .AddClass("content").Id("content")
                 .Add(S["Content Items"], S["Content Items"].PrefixPosition(), contentItems => contentItems
                     .Permission(Permissions.EditOwnContent)
-                    .Action(nameof(AdminController.List), typeof(AdminController).ControllerName(), new { area = "OrchardCore.Contents" })
+                    .Action(nameof(AdminController.List), typeof(AdminController).ControllerName(), new { area = "OrchardCore.Contents", contentTypeId = "" })
```

Comment:
This is agency theme.  Because this theme recipe doesn't define an admin menu you can see the issue solved by this changes in the two `AdminMenu.cs` classes. I have enabled Amin menu feature and create a link to a specific content type...the gif could explain better. 

After this example I believe is better to hide the "Content Type" filter on the right side because is not necessary when you are listing a specific content type and could cause strange behavior: I have put this PR in draft for this and I'm waiting about your approvation.

![agency_theme_issue](https://user-images.githubusercontent.com/2736009/92357141-a97eea00-f0e7-11ea-90db-78dadfb72614.gif)

Reply:
ok great gif, thank you.

I can see where the issue is now, where the blog recipe is adding a placeholder with no route values, i.e. `Admin/Contents/ContentItems/` - the Link route, which is overriding the one that is being produced in the agency theme.

I'll have a think about why that's happening, and if this is the best fix.

Reply:
> After this example I believe is better to hide the "Content Type" filter on the right side because is not necessary when you are listing a specific content type and could cause strange behavior: I have put this PR in draft for this and I'm waiting about your approbation.

About this, what do you think?
This could be the interface when a user click over a "type link"
![type_view](https://user-images.githubusercontent.com/2736009/92441848-393b9b80-f1af-11ea-9dba-3674a832ecd4.jpg)

And the current interface when a user click over a "generic link" (example Content Items)
![generic_view](https://user-images.githubusercontent.com/2736009/92441865-3f317c80-f1af-11ea-8fa0-4819cb779d83.jpg)


To distinguish between typed view and generic view I check `contentTypeId` : if is null or empty means the user want to list the items of all content types and eventually filter by type, if is not empty means the user want to list only the elements of the content type in the `contentTypeId`'s value. When user filter by content type, the view put the selected value in `model.Options.SelectedContentType`. 
If you approve it, I do the commit with the code to do this.

Reply:
> This is agency theme. Because this theme recipe doesn't define an admin menu you can see the issue solved by this changes in the two `AdminMenu.cs` classes. I have enabled Amin menu feature and create a link to a specific content type...the gif could explain better.
> 
> After this example I believe is better to hide the "Content Type" filter on the right side because is not necessary when you are listing a specific content type and could cause strange behavior: I have put this PR in draft for this and I'm waiting about your approbation.
> 
> ![agency_theme_issue](https://user-images.githubusercontent.com/2736009/92357141-a97eea00-f0e7-11ea-90db-78dadfb72614.gif)

@PiemP can you take this out of draft?

@jtkech if you have time do you mind a quick look? I can repro with the agency theme, and the fix looks ok (it's minor, adding the `contentTypeId = ""` to the routes), but I wouldn't mind to know if you think there might be an issue with the way menu item route values are merged (priorities), that is causing it, and I know you looked at them a while back

Reply:
> If you approve it, I do the commit with the code to do this.

This looks like it will be good, but can you do on another issue / pr after this please as I'd like to get this change through first?

Reply:
> This looks like it will be good, but can you do on another issue / pr after this please as I'd like to get this change through first?

ok,  no problem i create another issue and then another PR. Thank you

Reply:
@deanmarcussen 

> @jtkech if you have time do you mind a quick look?

Okay, I will do it this week end

Reply:
@deanmarcussen @PiemP after a quick look, adding the contentTypeId = "" to the routes LGTM

Didn't try it but i understant that if we set it with a non empty value for a given link, we then need to reset it when coming back to the list of all types, because the previous value may be part of the ambient values.

Reply:
In the `AdminMenu` of `OC.Menu` it also uses `Options.SelectedContentType` that is not part of the route pattern and then passed as a query parameter and that is not hold as an ambient value for a next link

            var rvd = new RouteValueDictionary
            {
                { "Area", "OrchardCore.Contents" },
                { "Options.SelectedContentType", "Menu" },
                { "Options.CanCreateSelectedContentType", true }
            };

That would become useless if we use `contentTypeId = "Menu"` that overrides `SelectedContentType` in the controller

Or maybe the opposite, let it as it was and then only update `ContentTypesAdminNodeNavigationBuilder` to use `SelectedContentType` (that will not be hold as an ambient value) in place of `contentTypeId`

                    cTypeMenu.Url(_linkGenerator.GetPathByRouteValues(_httpContextAccessor.HttpContext, "", new
                    {
                        area = "OrchardCore.Contents",
                        controller = "Admin",
                        action = "List",
                        // contentTypeId = ctd.Name // replace this one
                        Options.SelectedContentType = ctd.Name // by this one
                    }));

So we would not have to use `contentTypeId`. Hmm, but still keep `{contentTypeId?}` in the route pattern and as an action parameter to not break this route

Will see tomorrow, need more thinkings

Reply:
> So we would not have to use contentTypeId. Hmm, but still keep {contentTypeId?} in the route pattern and as an action parameter to not break this route

If you remove `contentTypeId` it could be difficult to understand when a user filter a content item list by content type (using the field on the right of the interface) and when a user click over a menu link that list elements of a specific content type. In my next PR I would hide the content type filter when a user click on a menu link that list elements of a specific content type. After my next PR `Options.SelectedContentType` could be used only when a user filter by content type a list of content item and `contentTypeId` could be used when a user want to list only a specific content type.

Hope to have write something that have sense.

> ...
> After this example I believe is better to hide the "Content Type" filter on the right side because is not necessary when you are listing a specific content type and could cause strange behavior: I have put this PR in draft for this and I'm waiting about your approbation.
> 
> ![agency_theme_issue](https://user-images.githubusercontent.com/2736009/92357141-a97eea00-f0e7-11ea-90db-78dadfb72614.gif)

Reply:
@PiemP 

Just for infos, by using this in `ContentTypesAdminNodeNavigationBuilder`

    cTypeMenu.Url(_linkGenerator.GetPathByRouteValues(_hca.HttpContext, "", new RouteValueDictionary
        {
            { "area", "OrchardCore.Contents" },
            { "controller", "Admin" },
            { "action", "List" },
            { "Options.SelectedContentType", ctd.Name },
            { "Options.CanCreateSelectedContentType", true }
        }));

It works as expected, the same view is used but with the right pre-selection of the content type. But if someone still uses a link with a specific `contentTypeId`, we would still have to reset this route data with the right value.

And i understand what you said, it makes sense. So i'm okay with always setting the `contentTypeId` route data. In that case, and if you also hide the content type filter, it would be useless to set `Options.SelectedContentType` e.g. in the `AdminMenu` of the `OC.Menu` module, hmm but we also use `Options.CanCreateSelectedContentType` that we check only if `Options.SelectedContentType` has a value, so maybe better to still set these 2 route values.

If so, to be consistent, in `ContentTypesAdminNodeNavigationBuilder` we would need to have

    cTypeMenu.Url(_linkGenerator.GetPathByRouteValues(_hca.HttpContext, "", new RouteValueDictionary
        {
            { "area", "OrchardCore.Contents" },
            { "controller", "Admin" },
            { "action", "List" },
            { "contentTypeId", ctd.Name },
            // As before but also with these 2 below values as done in OC.Menu
            { "Options.SelectedContentType", ctd.Name },
            { "Options.CanCreateSelectedContentType", true }
        }));

But not sure we want this by default

So as you want, i'm fine with what you said and want to do

Reply:
>If so, to be consistent, in ContentTypesAdminNodeNavigationBuilder we would need to have

>cTypeMenu.Url(_linkGenerator.GetPathByRouteValues(_hca.HttpContext, "", new RouteValueDictionary
    {
        { "area", "OrchardCore.Contents" },
        { "controller", "Admin" },
        { "action", "List" },
        { "contentTypeId", ctd.Name },
        // As before but also with these 2 below values as done in OC.Menu
        { "Options.SelectedContentType", ctd.Name },
        { "Options.CanCreateSelectedContentType", true }
    }));
>But not sure we want this by default

>So as you want, i'm fine with what you said and want to do

I'm not a good developer but I would like to do a work that you like. For my little experience I don't make `Options.SelectedContentType` and `Options.CanCreateSelectedContentType` by default because for `Options.SelectedContentType` is something managed by user interaction and, after my next PR, this parameter could only be set by the interface that list all the content items. For `Options.CanCreateSelectedContentType` I don't know: I don't have personally created this parameter and It's difficut to me understand the ratio of this parameter because it overwrite something in the content type definition. I'm pretty sure there is a good reason to introduce it but, about it, I believe is better to don't have a default value for this parameter because it's used only for the `Menu` link in the AdminMenu.

If you believe it's better to have default value, I prefer to have `Options.CanCreateSelectedContentType` set to `false` for the default one.

Reply:
@PiemP Okay cool, makes sense

So i suggest to not modify `ContentTypesAdminNodeNavigationBuilder`, I just wanted to be sure that everything is taken into consideration ;)